ttf2mesh.c

```diff
@@ -1151,18 +1157,23 @@ static int ttf_extract_tables(const uint8_t *data, int size, pps_t *s)
     rec = (ttf_tab_rec_t *)(s->hdr + 1);
 
     #define check_tag(str) (*(uint32_t *)rec->tableTag != *(uint32_t *)str)
-    #define match(type, name, str) \
+    #define match(type, name, str)                      \
     if (*(uint32_t *)rec->tableTag == *(uint32_t *)str) \
-    { \
-        s->s##name = rec->length; \
-        s->p##name = (type)(data + rec->offset); \
+    {                                                   \
+        s->s##name = rec->length;                       \
+        s->p##name = (type)(data + rec->offset);        \
+        if (rec->offset > INT_MAX - rec->length)        \
+            return TTF_ERR_FMT;                         \
+        if (rec->offset + rec->length > (unsigned)size) \
+            return TTF_ERR_FMT;                         \
```

Comment:
Checking that the table offset + length doesn't go over the input data length. Also added a integer overflow check (integer overflow occured where offset + length would wrap over)

Reply:
Thank you. Overflow checking is really important when dealing with potentially invalid files. I have paste validation before the matching and given that the size value is a priory less than TTF_MAX_FILE:
`        if (rec->offset > (uint32_t)size || rec->length > (uint32_t)size)
            return TTF_ERR_FMT;
`
`
        if (rec->offset + rec->length > (uint32_t)size) return TTF_ERR_FMT;
`